docs/tutorials/react18-migration.md

```diff
@@ -1,75 +1,127 @@
+:::info
+
+This guide has been updated to include steps for removing support for React 16, as it is now deprecated.
+
+:::
+
+The Backstage core libraries and plugins are compatible with all versions of React from v17 to v18. This means that you can migrate projects at your own pace. We do however encourage you to do so sooner rather than later, both to keep up with the evolving ecosystem, but also because React 18 brings performance improvements, in particular in tests.
 
 ## Migration
 
 _Before diving in, this is a heads-up that for large projects this can be a tricky migration due to the fact that it is hard to break down into a gradual migration. In practice the difficult part of this migration is switching to the new version of the `@testing-library/react` package for tests, since there is no overlapping support across major React versions, more on that later._
 
-### Switching to React 18
-
-To switch a project to React 18, there are generally three changes that need to be made.
-
-1. Update the resolutions in your root `package.json` to the new versions of `@types/react` and `@types/react-dom`:
-
-```json title="package.json"
-  "resolutions": {
-    // highlight-remove-next-line
-    "@types/react": "^17",
-    // highlight-remove-next-line
-    "@types/react-dom": "^17",
-    // highlight-add-next-line
-    "@types/react": "^18",
-    // highlight-add-next-line
-    "@types/react-dom": "^18",
-  },
-```
-
-2. Update the `react` and `react-dom` dependencies in your `packages/app/package.json` to the new versions:
-
-```json title="packages/app/package.json"
-  "dependencies": {
-    ...
-    // highlight-remove-next-line
-    "react": "^17.0.2",
-    // highlight-remove-next-line
-    "react-dom": "^17.0.2",
-    // highlight-add-next-line
-    "react": "^18.0.2",
-    // highlight-add-next-line
-    "react-dom": "^18.0.2",
-    ...
-  },
-```
-
-3. Update `packages/app/src/index.tsx` to use the new `react-dom/client` API to render the app:
-
-```tsx title="packages/app/src/index.tsx"
-import '@backstage/cli/asset-types';
-import React from 'react';
-// highlight-remove-next-line
-import ReactDOM from 'react-dom';
-// highlight-add-next-line
-import ReactDOM from 'react-dom/client';
-import App from './App';
-
-// highlight-remove-next-line
-ReactDOM.render(<App />, document.getElementById('root'));
-// highlight-add-next-line
-ReactDOM.createRoot(document.getElementById('root')!).render(<App />);
-```
-
-Once these steps are done you should be able to run your app and see it working as before, except now using React 18.
+### Upgrading to React 18
+
+#### Backstage Instance
+
+To migrate a Backstage instance to React 18, follow these three steps:
```

Comment:
I think we can just say follow these steps 😅, since reading it it is technically 3 + 1 at the end